kuma/javascript/src/header/dropdown.jsx

```diff
@@ -48,16 +53,46 @@ export default function Dropdown(props: DropdownProps) {
         });
     }
 
+    function hideDropdownMenuIfVisible() {
+        if (showDropdownMenu) {
+            setShowDropdownMenu(false);
+        }
+    }
+
+    useEffect(() => {
+        document.addEventListener('keyup', (event: KeyboardEvent) => {
+            if (event.key === 'Escape') {
+                hideDropdownMenuIfVisible();
+            }
+        });
+
+        document.addEventListener('click', (event: MouseEvent) => {
+            if (
+                event.target &&
+                event.target instanceof HTMLElement &&
+                !event.target.classList.contains('dropdown-menu-label')
+            ) {
+                hideDropdownMenuIfVisible();
+            }
+        });
```

Comment:
Would it be simpler to add an `onBlur` event to the button? Maybe something like
```
onBlur={() => {
  setShowDropdownMenu(false);
}}
```

Reply:
I don't believe so. When you use the keyboard and tab from the `button` to one of the dropdown menu items, it will fire the `blur` event and close the dropdown.

Reply:
Ah didn't think about that! Makes sense to keep as is then 👍